dowhy/causal_refuters/graph_refuter.py

```diff
@@ -1,69 +1,133 @@
 
 class GraphRefuter(CausalRefuter):
     """
     Class for performing refutations on graph and storing the results
     """
-    def __init__(self, data, method_name):
+    def __init__(self, data,method_name_discrete = "conditional_mutual_information",method_name_continuous = "partial_correlation"):
         """
         Initialize data for graph refutation
 
         :param data:input dataset
-        :param method_name: name of method for testing conditional independence
-
+        :param method_name_discrete: name of method for testing conditional independence in discrete data
+        :param method_name_continuous: name of method for testing conditional independece in continuous data
         :returns : instance of GraphRefutation class
         """
-        self._refutation_passed = False  
+        self._refutation_passed = "Undecided"  
         self._data = data
-        self._method_name = method_name
+        self._method_name_discrete = method_name_discrete
+        self._method_name_continuous = method_name_continuous
         self._false_implications = [] #List containing the implications from the graph which hold false for dataset
         self._true_implications = []  #List containing the implications from the graph which hold true for dataset
         self._results = {} #A dictionary with key as test set and value as [p-value, test_result]
         self.logger = logging.getLogger(__name__)
 
-    def set_refutation_result(self):
+    def set_refutation_result(self, number_of_constraints_model):
         """
         Method to set the result for graph refutation. Set true if there are no false implications else false
         """
-        if len(self._false_implications) == 0:
+        if (len(self._true_implications)) == number_of_constraints_model:
             self._refutation_passed = True
-        else:
+        elif len(self._false_implications) > 0:
             self._refutation_passed = False
+        else:
+            self._refutation_passed = "Undecided"
+    
+    def partial_correlation(self, x = None, y = None, z =None):
+        stats = partial_corr(data=self._data, x=x, y=y, z=list(z))
+        p_value = stats['p-val']
+        key = ((x,y)+(z,))
+        if(p_value < 0.05):
+            #Reject H0
+            self._false_implications.append([x,y,z])
+            self._results[key]= [p_value, False]
+        else:
+            self._true_implications.append([x, y, z])
+            self._results[key]= [p_value, True]
+
+    def conditional_mutual_information(self, x = None, y = None, z =None):
+         cmi_val = conditional_MI(data = self._data, x=x, y=y, z=list(z))
+         key = ((x,y)+(z,))
+         if(cmi_val <= 0.05):
+            self._true_implications.append([x, y, z])
+            self._results[key]= [cmi_val, True]
+         else:
+            self._false_implications.append([x,y,z])
+            self._results[key]= [cmi_val, False]
       
     def refute_model(self,independence_constraints):
```

Comment:
this docstring is missing the ":returns:" line

Reply:
This function returns nothing. Should I still add returns ?

Reply:
I meant this comment for `refute_model` method. that returns the `refute` variable, right?

Reply:
okay, yes